user_linux_cgo.go

```diff
@@ -0,0 +1,51 @@
+// +build cgo,linux
+
+package buildah
+
+// #include <pwd.h>
+// #include <stdlib.h>
+// #include <stdio.h>
+// #include <string.h>
+import "C"
+
+import (
+	"path/filepath"
+	"sync"
+	"unsafe"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func getUser(rootdir, username string) (specs.User, error) {
```

Comment:
Ok, showing my golang ignorance here.  What does specs.user and error in the second parens do?